projects/plugins/jetpack/_inc/polldaddy-shortcode.js

```diff
@@ -1,74 +1,94 @@
 
-					if ( poll ) {
-						var poll_url = document.createElement( 'a' );
-						poll_url.href = poll[ 'url' ];
-						if (
-							poll_url.hostname != 'secure.polldaddy.com' &&
-							poll_url.hostname != 'static.polldaddy.com'
-						) {
-							return false;
-						}
-						var pathname = poll_url.pathname;
-						if ( ! /\/?p\/\d+\.js/.test( pathname ) ) {
-							return false;
-						}
-						var wp_pd_js = document.createElement( 'script' );
-						wp_pd_js.type = 'text/javascript';
-						wp_pd_js.src = poll[ 'url' ];
-						wp_pd_js.charset = 'utf-8';
-						wp_pd_js.async = true;
-						document.getElementsByTagName( 'head' )[ 0 ].appendChild( wp_pd_js );
+				if ( isInitialized ) {
+					return;
+				}
+
+				pollEl.setAttribute( 'data-pd-init-done', '1' );
+				const settings = pollEl.getAttribute( 'data-settings' );
+				const poll = settings ? JSON.parse( settings ) : null;
+
+				if ( poll ) {
+					let poll_url;
+					try {
+						poll_url = new URL( poll.url, 'https://invalid.tld' );
+					} catch ( error ) {
+						return false;
 					}
-				} );
-			}
+					if (
+						poll_url.hostname !== 'secure.polldaddy.com' &&
+						poll_url.hostname !== 'static.polldaddy.com'
+					) {
+						return false;
+					}
+					const pathname = poll_url.pathname;
+					if ( ! /\/?p\/\d+\.js/.test( pathname ) ) {
+						return false;
+					}
+					const wp_pd_js = d.createElement( 'script' );
+					wp_pd_js.src = poll.url;
+					wp_pd_js.async = true;
+					d.head.appendChild( wp_pd_js );
+				}
+			} );
 
-			if ( ratings ) {
-				var script = '';
+			if ( ratings.length ) {
+				let scriptContents = '';
 
-				$.each( ratings, function () {
-					var rating = $( this ).data( 'settings' );
+				ratings.forEach( ratingEl => {
+					const isInitialized = ratingEl.getAttribute( 'data-pd-init-done' );
+
+					if ( isInitialized ) {
+						return;
+					}
+
+					ratingEl.setAttribute( 'data-pd-init-done', '1' );
+
+					const settings = ratingEl.getAttribute( 'data-settings' );
+					const rating = settings ? JSON.parse( settings ) : null;
 
 					if ( rating ) {
-						script +=
-							'PDRTJS_settings_' +
-							rating[ 'id' ] +
-							rating[ 'item_id' ] +
-							'=' +
-							rating[ 'settings' ] +
-							"; if ( typeof PDRTJS_RATING !== 'undefined' ){ if ( typeof PDRTJS_" +
-							rating[ 'id' ] +
-							rating[ 'item_id' ] +
-							"=='undefined' ){PDRTJS_" +
-							rating[ 'id' ] +
-							rating[ 'item_id' ] +
-							'= new PDRTJS_RATING( PDRTJS_settings_' +
-							rating[ 'id' ] +
-							rating[ 'item_id' ] +
-							' );}}';
+						scriptContents += `
+						PDRTJS_settings_${ rating.id }${ rating.item_id } = ${ rating.settings };
+						if ( typeof PDRTJS_RATING !== 'undefined' ) {
+							if ( typeof PDRTJS_${ rating.id }${ rating.item_id } === 'undefined' ) {
+								PDRTJS_${ rating.id }${ rating.item_id } =
+									new PDRTJS_RATING( PDRTJS_settings_${ rating.id }${ rating.item_id } );
+							}
+						}
+						`;
```

Comment:
This code is kind of ugly, but I guess we don't have to fix it now. Your changed version here looks like it will do the same as the previous version.

Reply:
Right, my goal was merely to make it more readable, by moving it to a string template. I don't have enough confidence to go about rewriting it, because of the interaction it has with other scripts that get loaded.